components/infra-proxy-service/service/service.go

```diff
@@ -48,7 +48,7 @@ func ParseStorageError(err error, v interface{}, noun string) error {
 		case storage.ErrCannotDelete:
 			return status.Errorf(codes.FailedPrecondition, "cannot delete server %q because it still has organizations attached", reflect.Indirect(reflect.ValueOf(v)).FieldByName("Id"))
 		case storage.ErrConflict:
-			return status.Errorf(codes.AlreadyExists, "%s with that name %q already exists", noun, reflect.Indirect(reflect.ValueOf(v)).FieldByName("Name"))
+			return status.Errorf(codes.AlreadyExists, "%s with ID %q already exists", noun, reflect.Indirect(reflect.ValueOf(v)).FieldByName("Id"))
```

Comment:
Wondering why you need all the reflection on these? Is that used to make this more generic?

Reply:
Yeah, I have generalized it up to some extent, as we have fixed the issue "More specific foreign_key_violation error messages" in PR https://github.com/chef/automate/pull/3298.

checking of any improvement it can put here... Thanks!